Issue #5: Add capability requirement for access key creation

tomdickman opened the pull request.

Commit:
Issue #5: Add capability requirement for access key creation

dmitriim commented on classes/helper.php:
```diff
@@ -65,12 +65,13 @@ class helper {
-        if (empty($USER->id)) {
-            throw new \coding_exception('Cannot create a user key when not logged in as a user.');
+        if (!has_capability('tool/pdfpages:createaccesskey', \context_system::instance())) {
+            throw new \moodle_exception('error:permissions:createkey', 'tool_pdfpages');
```
you can replace that with require_capability(\context_system::instance()), 'tool/pdfpages:createaccesskey', 'error:permissions:createkey', 'tool_pdfpages')

tomdickman replied:
Replaced as suggested

dmitriim commented on README.md:
```diff
@@ -43,6 +43,8 @@ git clone git@github.com:catalyst/moodle-tool_pdfpages.git <moodledir>/admin/too
 
 Use of the converter requires programmatic access, there in no frontend associated with this plugin, so you need to develop another module, or add this plugin to the dependencies of an existing Moodle plugin.
 
+> Only users with the system level capability `tool/pdfpages:createaccesskey` can conducted conversions, as this is required to create the single use access key for the headless browser session.
```
how that would work if we would like to get users to generate pdf files by clicking an link in the browser? 


From the business point of view  pdfpages:createaccesskey does mean nothing. They shouldn't be aware about what's happening under the hood. I think we should rather call this cap something like "generate pdf from URL"  which will make more sense.

dmitriim replied:
Also it feels like we should check that permissions in index.php ?

tomdickman replied:
> how that would work if we would like to get users to generate pdf files by clicking an link in the browser?

This plugin API isn't designed for this use case at the moment, it is designed for the conversion to be offloaded to a scheduled and/or adhoc task and the link should serve the converted file once conversion is complete. So if live conversion was to occur, the user would either require the capability to generate PDFs, or conversion would need to be offloaded to another process (like an adhoc task) otherwise this would be an enhancement and there may be security concerns around allowing any user to click on a link and conduct conversion immediately.

> From the business point of view pdfpages:createaccesskey does mean nothing. They shouldn't be aware about what's happening under the hood. I think we should rather call this cap something like "generate pdf from URL" which will make more sense.

Agreed, amended capability to more accurately reflect the purpose.

tomdickman replied:
> Also it feels like we should check that permissions in index.php ?

Added a capability check to proxy following user login

Commit:
Issue #5: Rename capability and add check to proxy

Rename the capability to more accurately reflect what it enables
user to do.
Check capability when creating the access key and again in proxy
for security.

Commit:
Issue #5: Amend README to reflect capability name change.

dmitriim commented on lang/en/tool_pdfpages.php:
```diff
@@ -26,10 +26,11 @@
-$string['error:invalidpageoption'] = 'The PDF page option you selected is not supported: {$a}';
 $string['error:converternotfound'] = 'Could not find enabled converter, please check tool_pages plugin settings.';
-$string['error:wkhtmltopdfpath'] = 'Cannot establish path to wkhtmltopdf, please check setting tool_pdfpages|wkhtmltopdfpath.';
+$string['error:invalidpageoption'] = 'The PDF page option you selected is not supported: {$a}';
+$string['error:permissions:createkey'] = "User doesn't have required capability to create access keys.";
```
Where do we use these strings?

Commit:
Issue #5: Remove unused string
